modules/carto/src/api/parseMap.js

```diff
@@ -24,8 +24,10 @@ export function parseMap(json) {
     mapStyle,
     layers: extractTextLayers(layers.reverse()).map(({id, type, config, visualChannels}) => {
       try {
-        log.assert(type in LAYER_MAP, `Unsupported layer type: ${type}`);
-        const {Layer, propMap, defaultProps} = LAYER_MAP[type];
+        const layer = getLayer(type, config);
+
+        log.assert(getLayer(type, config), `Unsupported layer type: ${type}`);
```

Comment:
Move this assertion into `getLayer` to avoid double calling the function?

Reply:
Ups, thank you!